test/cpp/test_aten_xla_tensor.cpp

```diff
@@ -20,629 +16,622 @@ namespace cpp_test {
 class AtenXlaTensorTest : public AtenXlaTensorTestBase {};
 
 void TestBackward(
-    const std::vector<at::Tensor>& inputs, const Device& device,
-    const std::function<at::Tensor(const std::vector<at::Tensor>&)>& testfn,
-    double rtol = 1e-5, double atol = 1e-8,
-    const std::vector<bool>& inputs_require_grad = {}) {
-  CHECK(inputs_require_grad.empty() ||
-        inputs.size() == inputs_require_grad.size());
-  std::vector<at::Tensor> input_vars;
-  std::vector<at::Tensor> xinput_vars;
+    const std::vector<torch::Tensor>& inputs, const torch::Device& device,
+    const std::function<torch::Tensor(const std::vector<torch::Tensor>&)>& testfn,
+    double rtol = 1e-5, double atol = 1e-8) {
+  std::vector<torch::Tensor> input_vars;
+  std::vector<torch::Tensor> xinput_vars;
   for (size_t i = 0; i < inputs.size(); ++i) {
     auto& input = inputs[i];
     if (input.defined()) {
-      const bool requires_grad =
-          inputs_require_grad.empty() ? true : inputs_require_grad[i];
-      input_vars.push_back(
-          torch::autograd::make_variable(input, requires_grad));
-
-      at::Tensor xinput = bridge::CreateXlaTensor(CopyTensor(input), device);
-      xinput_vars.push_back(
-          torch::autograd::make_variable(xinput, requires_grad));
+      input_vars.push_back(input.clone().detach().set_requires_grad(input.requires_grad()));
+
+      torch::Tensor xinput = input.to(device, /*non_blocking=*/false, /*copy=*/true).detach().set_requires_grad(input.requires_grad());
+      xinput_vars.push_back(xinput);
     } else {
       input_vars.emplace_back();
       xinput_vars.emplace_back();
     }
   }
 
-  at::Tensor output = testfn(input_vars);
-  at::Tensor xoutput = testfn(xinput_vars);
+  torch::Tensor output = testfn(input_vars);
+  torch::Tensor xoutput = testfn(xinput_vars);
   output.backward();
   xoutput.backward();
-  for (size_t i = 0; i < input_vars.size(); ++i) {
-    if (inputs[i].defined() && input_vars[i].requires_grad()) {
+  for (size_t i = 0; i < inputs.size(); ++i) {
+    if (inputs[i].defined() && inputs[i].requires_grad()) {
       ASSERT_TRUE(xinput_vars[i].grad().defined());
       AllClose(input_vars[i].grad(), xinput_vars[i].grad(), rtol, atol);
     }
   }
 }
 
 TEST_F(AtenXlaTensorTest, TestScalarTensor) {
```

Comment:
I am assuming that from this point forward, within this file, we have only trivial search&replace changes, right?

Reply:
Yep that's correct.